test/unit/events.js

```diff
@@ -201,3 +202,23 @@ test('should have a defaultPrevented property on an event that was prevent from
 
   Events.trigger(el, 'test');
 });
+
+test('should have relatedTarget correctly set on the event', function() {
+  expect(2);
+
+  var el1 = document.createElement('div'),
+      el2 = document.createElement('div'),
+      relatedEl = document.createElement('div');
+
+  Events.on(el1, 'click', function(e){
+    equal(e.relatedTarget, relatedEl, 'relatedTarget is set for all browsers when referring element is set on the event');
+  });
+
+  Events.trigger(el1, TestHelpers.makeMouseEvent('click', relatedEl));
```

Comment:
If we're using Events.trigger I think we could just use a plain object here instead of creating a MouseEvent. We end up [rebuilding it as an object](https://github.com/videojs/video.js/blob/master/src/js/events.js#L31) either way. e.g. `{ type: 'click', relatedTarget: relatedEl }`